docs/csharp/language-reference/statements/jump-statements.md

```diff
@@ -0,0 +1,114 @@
+
+- Without expression, as in the following example:
+
+  :::code language="csharp" interactive="try-dotnet-method" source="snippets/jump-statements/ReturnStatement.cs" id="WithoutExpression":::
+
+  You can use the `return` statement without expression in a function that doesn't compute a value, that is:
+
+  - a [method](../../programming-guide/classes-and-structs/methods.md), [local function](../../programming-guide/classes-and-structs/local-functions.md), or [lambda expression](../operators/lambda-expressions.md) with the result type [`void`](../builtin-types/void.md)
+  - an [async](../keywords/async.md) method, local function, or lambda expression with the result type <xref:System.Threading.Tasks.Task> or <xref:System.Threading.Tasks.ValueTask>
+  - the `set` accessor of a [property](../../programming-guide/classes-and-structs/properties.md) or an [indexer](../../programming-guide/indexers/index.md)
+  - a [constructor](../../programming-guide/classes-and-structs/constructors.md)
+  - a [finalizer](../../programming-guide/classes-and-structs/finalizers.md)
+  - the `add` and `remove` accessors of an [event](../../programming-guide/events/index.md)
+
+  You typically use the `return` statement without expression to terminate a function early. If a function doesn't contain the `return` statement, it terminates after its last statement is executed.
+
+- With expression, as in the following example:
+
+  :::code language="csharp" interactive="try-dotnet-method" source="snippets/jump-statements/ReturnStatement.cs" id="WithExpression":::
+
+  You use the `return` statement with expression in a function that computes a value, that is:
+
+  - a [method](../../programming-guide/classes-and-structs/methods.md), [local function](../../programming-guide/classes-and-structs/local-functions.md), or [lambda expression](../operators/lambda-expressions.md) with a non-void result type
+  - an [async](../keywords/async.md) method, local function, or lambda expression with the result type <xref:System.Threading.Tasks.Task%601> or <xref:System.Threading.Tasks.ValueTask%601>
+  - the `get` accessor of a [property](../../programming-guide/classes-and-structs/properties.md) or an [indexer](../../programming-guide/indexers/index.md)
+  - an [overloaded operator](../operators/operator-overloading.md)
+
+  By default, the `return` statement returns the value of an expression. Beginning with C# 7.0, you can return a reference to a variable. To do that, use the `return` statement with the [`ref` keyword](../keywords/ref.md), as the following example shows:
+
+  :::code language="csharp" interactive="try-dotnet-method" source="snippets/jump-statements/ReturnStatement.cs" id="RefReturn":::
+
+  For more information about ref returns, see [Ref returns and ref locals](../../programming-guide/classes-and-structs/ref-returns.md).
+
+## The `goto` statement
```

Comment:
I think it's worth mentioning that there are restrictions on the target label of a `goto` statement.

The [standard](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/statements.md#13104-the-goto-statement) has some very elegant language to define the restriction:

> If a label with the given name does not exist in the current function member, or if the `goto` statement is not within the scope of the label, a compile-time error occurs.

This rule means you can't use `goto` to transfer control *into* a `try` block, `catch` block, `finally` block, or other nested scope.

It might be sufficient to say you can't use `goto` to transfer control out of the current member, or into any nested scope, such as a `try` block.

Reply:
@BillWagner thanks, I've updated the section accordingly.